SpatialGDK/Source/SpatialGDK/Private/Interop/SkeletonEntities.cpp

```diff
@@ -0,0 +1,485 @@
+
+			UE_LOG(LogSpatialSkeletonEntityCreator, Log,
+				   TEXT("EntityId: %lld Delegating skeleton entity to VirtualWorker %d Partition %lld"), CreatedSkeletonEntityId,
+				   AuthWorkerId, AuthWorkerPartition);
+
+			AuthorityDelegation AuthorityDelegationComponent;
+			AuthorityDelegationComponent.Delegations.Emplace(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID, AuthWorkerPartition);
+
+			EntitiesToPopulatingWorkers.Emplace(CreatedSkeletonEntityId, AuthWorkerId);
+
+			NetDriver->Connection->GetCoordinator().SendComponentUpdate(
+				CreatedSkeletonEntityId,
+				ComponentUpdate(OwningComponentUpdatePtr(AuthorityDelegationComponent.CreateAuthorityDelegationUpdate().schema_type),
+								AuthorityDelegation::ComponentId),
+				{});
+		}
+
+		SkeletonEntitiesToDelegate.Empty();
+
+		UE_LOG(LogSpatialSkeletonEntityCreator, Log, TEXT("Finished delegating entities, moving on to manifest"));
+
+		Stage = EStage::SigningManifest;
+	}
+
+	if (Stage == EStage::SigningManifest)
+	{
+		TMap<VirtualWorkerId, TSet<Worker_EntityId_Key>> WorkerToEntitiesToPopulate;
+		// Every worker should get a manifest to proceed through the skeleton entity flow.
+		for (const auto& VirtualWorker : NetDriver->WellKnownEntitySystem->VirtualWorkerTranslationManager->GetVirtualWorkerMapping())
+		{
+			WorkerToEntitiesToPopulate.FindOrAdd(VirtualWorker.Key);
+		}
+		for (const auto& SkeletonEntityToPopulatingWorker : EntitiesToPopulatingWorkers)
+		{
+			WorkerToEntitiesToPopulate.FindOrAdd(SkeletonEntityToPopulatingWorker.Value).Add(SkeletonEntityToPopulatingWorker.Key);
+		}
+		for (const auto& WorkerToEntitiesToPopulateEl : WorkerToEntitiesToPopulate)
+		{
+			const VirtualWorkerId WorkerId = WorkerToEntitiesToPopulateEl.Key;
+			const TSet<Worker_EntityId_Key>& EntitiesToPopulate = WorkerToEntitiesToPopulateEl.Value;
+
+			TArray<ComponentData> SkeletonEntityManifestComponents;
+
+			{
+				FSkeletonEntityManifest Manifest;
+				Manifest.EntitiesToPopulate = EntitiesToPopulate;
+
+				SkeletonEntityManifestComponents.Emplace(Manifest.CreateComponentData());
+			}
+
+			const Worker_PartitionId ServerWorkerPartitionId =
+				NetDriver->VirtualWorkerTranslator->GetPartitionEntityForVirtualWorker(WorkerId);
+
+			{
+				const Worker_PartitionId ThisWorkerPartitionId = NetDriver->VirtualWorkerTranslator->GetClaimedPartitionId();
+				AuthorityDelegation Authority;
+				Authority.Delegations.Emplace(SpatialConstants::SKELETON_ENTITY_MANIFEST_AUTH_COMPONENT_SET_ID, ServerWorkerPartitionId);
+				Authority.Delegations.Emplace(SpatialConstants::SPATIALOS_WELLKNOWN_COMPONENTSET_ID, ThisWorkerPartitionId);
+
+				SkeletonEntityManifestComponents.Emplace(
+					ComponentData(OwningComponentDataPtr(Authority.CreateComponentData().schema_type), AuthorityDelegation::ComponentId));
+			}
+
+			{
+				Query Q;
+				Q.Constraint.bSelfConstraint = true;
+				Q.ResultComponentIds.Append({ SpatialConstants::SKELETON_ENTITY_MANIFEST_COMPONENT_ID });
```

Comment:
Maybe make the GSM interested in all manifests somehow, removing the need for SPATIALOS_WELLKNOWN_COMPONENTSET_ID auth & leaving less queries?

Reply:
The GSM being interested probably doesn't help us

Reply:
It does seem to - the GSM-auth worker will receive interest over all the manifests, seems to achieve the same as specifying GSM-auth worker to be auth over WELL_KNOWN component set on every manifest.